### PR TITLE
Removing pyaudio fixing failures

### PR DESCRIPTION
**Pull request by oplatek**
PR:
Pros:
- strips old code
- fixes few test failures
- starts with 0 tests failing

Cons:
- This pull request uses new pykaldi interface (which is fixed in separate branch - 
- Few tests were disabled see nose2.cfg
- Most of the library is untested
- The Alex library depends on lot of libraries e.g. pjsip, pykaldi and python libraries all build from source using 
  only python2.7 and gcc of 1 version see travis.yml for details. Other compilers/python versions are untested.
- Pykaldi is even prebuild also using travis. The prebuild Pykaldi (openfst library & Python eggs for Pykaldi and Pyfst) are stored in S3 storage which belongs to @ticcky (probably it would be worth to change it the future so he will not pay for the service)

### CODE REVIEW

File: alex/tools/htk/bin/export_models.sh [hunk 1]
```diff
@@ -33,5 +33,5 @@ if [ "$2" = "text" ]; then
 fi
 
 # Create Julius ASR binary AM models.
```
Comment (jurcicek) [inline alex/tools/htk/bin/export_models.sh]: Please remove it completely. We will no work with Julius any more.

**Comment by jurcicek** [comment]
I left one comment. Otherwise it looks good to me.

File: .travis.yml [hunk 2]
```diff
@@ -14,20 +14,22 @@ compiler:
...
     - echo "#####################"; echo "vvv PIP LOG vvv"; tail -n 100 /tmp/pip_log.log ; echo "#####################";
     # Install pjsip.
     - (git clone https://github.com/UFAL-DSG/pjsip.git && pushd pjsip && git checkout travis && ./configure CXXFLAGS=-fPIC CFLAGS=-fPIC LDFLAGS=-fPIC CPPFLAGS=-fPIC && make dep && make && sudo make install && cd pjsip-apps/src/python/ && pip install . && popd && python -c 'import pjsuaxt') &> /dev/null
     # Unpack and install precomplied openfst, pykaldi and pyfst
-    - (wget https://s3-eu-west-1.amazonaws.com/ufal-dsg/pykaldi/pykaldi-travis_2.7.tar.gz/pykaldi/pykaldi_Ubuntu_12.04_LTS.tar.gz && tar xf pykaldi_Ubuntu_12.04_LTS.tar.gz && pushd pykaldi_Ubuntu_12.04_LTS && sudo cp -r openfst/lib/* /usr/local/lib && sudo ldconfig && sudo tar xf pyfst*.tar -C / && sudo tar xf pykaldi*.tar -C / ; popd ) &> /dev/null
+    - wget https://s3-eu-west-1.amazonaws.com/ufal-dsg/pykaldi/pykaldi-travis_2.7.tar.gz/pykaldi_Ubuntu_12.04_LTS.zip && unzip pykaldi_Ubuntu_12.04_LTS.zip && pushd pykaldi_Ubuntu_12.04_LTS && sudo cp -r openfst/lib/* /usr/local/lib && easy_install pyfst*.egg && easy_install pykaldi*.egg && sudo ldconfig && popd
```
Comment (ticcky) [inline .travis.yml]: When you keep the parentheses ( ) around the command you can leave out the trickery with pushd and popd =)

**Comment by ticcky** [comment]
Why did you remove the following?
test_category_labels_substitution.py
alex/components/hub/webio.py
alex/components/hub/ahub.py
alex/components/hub/aio.py
alex/utils/audio_play.py

Has it been decided that the only interface we will be supporting is the VoIP?

File: .travis.yml (hunk 2 above)
Comment (oplatek) [inline .travis.yml]: Well, I have mainly removed the redirection to /dev/null 
because  the wget is quite likely step to fail

**Comment by jurcicek** [comment]
Regarding, Voip

we will longer support desktop audio. This is not useful and probably it
does not work as of now.


Best regards,
Filip Jurcicek

-------------------------------------------------------
Work tel. (CZ): +420221914402
Personal tel. (CZ): +420777805048
Skype: bozskyfilip

http://ufal.mff.cuni.cz/filip-jurcicek
http://ufal.mff.cuni.cz/grants

**Comment by oplatek** [comment]
@ticcky 

As Filip commented: the desktop audio is not very much used and maintained.
So I remove the components using the pyaudio:
alex/components/hub/webio.py
alex/components/hub/ahub.py
alex/components/hub/aio.py
alex/utils/audio_play.py

The pyaudio is quite heavy dependency.
(I plan to adapt slowly create web based for Alex,
so there will be also interface probably using SpeechRecognition.js from cloud-asr)

About test_category_labels_substitution.py
It is so outdated that it is worth to write new test than fixing this one.

**Comment by oplatek** [comment]
@jurcicek I removed the htk/bin/export_models.sh

**Comment by ticcky** [comment]
I'm not sure why is it not useful, but I agree that we don't know whether it works now =) However, considering that somebody might want to run Alex on Raspberry Pi like Jasper [1] and webio.py is I think immensely useful when debugging the system, I don't think we should just stop supporting them with no alternative.

Let's unwind pyaudio from the source code so that for voip functionality we do not need to have it installed. And put pyaudio into optional dependencies. But I don't think we should just remove this.

[1] http://cmusphinx.sourceforge.net/2014/04/jasper-personal-assistant-for-raspberry-pi/

**Comment by oplatek** [comment]
@ticcky @jurcicek I updated the pull request so it keeps webhub and ahub

**Pull request merged**